pkg/azure/ocpgwdeployer.go

```diff
@@ -79,40 +79,36 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
 
 	status.Start("Deploying gateway node")
 
-	nsgClient, err := d.getNsgClient()
+	nsgClient, nwClient, pubIPClient, err := d.getClients(status)
 	if err != nil {
-		return status.Error(err, "Failed to get network security groups client")
+		return status.Error(err, "Failed to get client")
```

Comment:
`getClients` logs errors.

```suggestion
		return err
```